backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/document/model/impl/ElasticsearchIndexSchemaFieldNode.java

```diff
@@ -43,15 +51,16 @@ public ElasticsearchFieldPredicateBuilderFactory getPredicateBuilderFactory() {
 	}
 
 	public boolean isCompatibleWith(ElasticsearchIndexSchemaFieldNode<?> other) {
-		return Objects.equals( codec, other.codec )
-				&& Objects.equals( predicateBuilderFactory, other.predicateBuilderFactory );
+		return converter.isDslCompatibleWith( other.converter )
+				&& Objects.equals( codec, other.codec )
```

Comment:
Can the codec here be different from the one in the converter?

Because we already test the codec in the converter when doing `converter.isDslCompatibleWith( other.converter )`.

What I'm also wondering is if it makes sense to still have a codec field here if it is included in the converter. I would keep the `getCodec()` getter though.